types/baseui/baseui-tests.tsx

```diff
@@ -111,6 +111,8 @@ const NewStyledProgressSteps = styled<ProgressStepsStyleProps, typeof StyledProg
 // Avatar
 <Avatar name='user' size='10'/>;
 <Avatar name='user' size={10}/>; // $ExpectError
+<Avatar name="user" overrides={{ Root: { style: ({$theme}) => ({height: $theme.sizing && $theme.sizing.notasize})} }} />; // $ExpectError
+<Avatar name="user" overrides={{ Root: { style: ({$nonExistantProp}) => ({}) }}} />; // $ExpectError
```

Comment:
Maybe an extra test here that verifies correct params are allowed too.
`<Avatar name="user" overrides={{ Root: { style: ({$size, role, children}) => ({}) }}} />;`

Reply:
@jh3y I added happy path tests for the the $theme and the passed in shared props. 

I have a question about `role` and `children` as these aren't included in the StylePropsT type and so not available to the style function in this implementation.  Should those props be added to the generic [here](https://github.com/uber-web/baseui/blob/cc272a2d999f6d39cccde26087113955101bd32e/src/avatar/types.js#L29) or should React intrinsic props, children for example, be defined in the style function type along with $theme?

Reply:
Hey @alexzywiak 👋 

Yeah, this is kinda what I was getting at. Because these are the props that would be available in that function and they might be used. For example, styling might change dependant on the number of children.

I think maybe the latter would make more sense.

Reply:
@jh3y, made a change to include React children in props.  Two issues I'm not sure how to approach:

1.  Additional props that are put directly onto the root component.  `role` and `aria-label` in Avatar, `onClick` in Button.  For Avatar, `role` would be available for Root overrides, but not Initials.
1. The `style` and the `component` override have access to the value of the `props` override key.

These are beyond my typescript prowess to figure out at the moment.  I think this style function interface is an improvement over `props: any`, but it's still not accurate.  Let me know what you want to do with it.

Reply:
Number 1. is easy once I thought about it.  Just change the Root override generic in Avatar specifically.

Reply:
I have approved the PR @alexzywiak 👍 
If you're happy with it and it won't break other definitions whilst making this one more accurate.